services/datamanager/builtin/file_deletion_test.go

```diff
@@ -0,0 +1,183 @@
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"math"
+	"os"
+	"sync/atomic"
+	"testing"
+
+	v1 "go.viam.com/api/app/datasync/v1"
+	"go.viam.com/test"
+
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/services/datamanager/datasync"
+)
+
+func TestFileDeletionUsageCheck(t *testing.T) {
+	tests := []struct {
+		name                 string
+		expectedFsCheckValue bool
+		testThreshold        float64
+		testCaptureDirRatio  float64
+		captureDirExists     bool
+	}{
+		{
+			name:                 "if not at file system capactiy threshold, we should return false from deletion check",
+			expectedFsCheckValue: false,
+			testThreshold:        .99,
+			testCaptureDirRatio:  .99,
+			captureDirExists:     true,
+		},
+		{
+			name:                 "if at file system capactiy threshold, we return true from deletion check",
+			expectedFsCheckValue: true,
+			testThreshold:        math.SmallestNonzeroFloat64,
+			testCaptureDirRatio:  math.SmallestNonzeroFloat64,
+			captureDirExists:     true,
+		},
+		{
+			name: "if at file system capactiy threshold but not capture dir threshold," +
+				"we return false from deletion check",
+			expectedFsCheckValue: false,
+			testThreshold:        math.SmallestNonzeroFloat64,
+			testCaptureDirRatio:  1.0,
+			captureDirExists:     true,
+		},
+		{
+			name:                 "if capture dir does not exist, we should return false from deletion check",
+			expectedFsCheckValue: false,
+			testThreshold:        .95,
+			testCaptureDirRatio:  .5,
+			captureDirExists:     false,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			var tempCaptureDir string
+			if tc.captureDirExists {
+				tempCaptureDir = t.TempDir()
+				// overwrite thresholds
+				fsThresholdToTriggerDeletion = tc.testThreshold
+				captureDirToFSUsageRatio = tc.testCaptureDirRatio
+				// write testing files
+				writeFiles(t, tempCaptureDir, false)
+			} else {
+				// make a random dir name to make sure it doesn't exist
+				randomName := make([]byte, 4)
+				_, err := rand.Read(randomName)
+				test.That(t, err, test.ShouldBeNil)
+				tempCaptureDir = hex.EncodeToString(randomName)
+			}
+
+			logger := logging.NewTestLogger(t)
+			willDelete, err := shouldDeleteBasedOnDiskUsage(context.Background(), tempCaptureDir, logger)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, willDelete, test.ShouldEqual, tc.expectedFsCheckValue)
+		})
+	}
+}
+
+func TestFileDeletion(t *testing.T) {
+	tests := []struct {
+		name                 string
+		syncEnabled          bool
+		markAllInProgress    bool
+		writeProgFiles       bool
+		shouldCancelContext  bool
+		expectedDeletedCount int
+	}{
+		{
+			name:                 "deletion with sync disabled should delete every 4th file",
+			expectedDeletedCount: 3,
+		},
+		{
+			name:                 "deletion with sync enabled and all files marked as in progress should not delete any files",
+			syncEnabled:          true,
+			markAllInProgress:    true,
```

Comment:
It was intially serving a different purpose that is now useless. Ill change it to a test that marks some but not all and make sure deleted is lower but not 0.

Reply:
Instead of how many files are marked in progress being hidden in the tests, maybe the number of files to be marked in progress should be written here or documented somewhere at the top-level instead of within the test implementation.